openwisp_firmware_upgrader/admin.py

```diff
@@ -200,9 +234,30 @@ def has_delete_permission(self, request, obj):
     def has_add_permission(self, request, obj):
         return False
 
+    class Media:
+        css = {'all': ['firmware-upgrader/css/upgrade-options.css']}
+
+
+class ReadonlyUpgradeOptionsMixin:
+    @admin.display(description=_('Upgrade options'))
+    def readonly_upgrade_options(self, obj):
+        # TODO: The OpenWrt schema is hard coded here. We need to find
+        # a way to dynamically select schema of the appropriate upgrader.
+        options = []
+        for key, value in OpenWrt.SCHEMA['properties'].items():
```

Comment:
3, these lines need to call a function or something which centralizes the logic to figure out which schema to use.